benchmark/storage/index_wrapper_benchmark.cpp

```diff
@@ -0,0 +1,154 @@
+#include <memory>
+#include <vector>
+
+#include "benchmark/benchmark.h"
+#include "common/scoped_timer.h"
+#include "parser/expression/column_value_expression.h"
+#include "storage/garbage_collector_thread.h"
+#include "storage/index/index.h"
+#include "storage/index/index_builder.h"
+#include "storage/projected_row.h"
+#include "storage/sql_table.h"
+#include "type/type_id.h"
+#include "util/catalog_test_util.h"
+#include "util/multithread_test_util.h"
+
+namespace terrier {
+
+class IndexBenchmark : public benchmark::Fixture {
+ private:
+  const std::chrono::milliseconds gc_period_{10};
+  storage::GarbageCollector *gc_;
+  storage::GarbageCollectorThread *gc_thread_;
+
+  storage::BlockStore block_store_{1000, 1000};
+  storage::RecordBufferSegmentPool buffer_pool_{1000000, 1000000};
+  catalog::Schema table_schema_;
+  catalog::IndexSchema index_schema_;
+
+ public:
+  std::default_random_engine generator_;
+
+  const uint32_t table_size_ = 10000000;
```

Comment:
It might actually make sense to add another 0 (make it 1e8) to the end of this number for the benchmark machines since they have 15MB of L3. I was really only worried about my laptop at the time I was profiling.